api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto

```diff
@@ -295,7 +295,19 @@ message ExtAuthz {
   // Additional metadata to be added to the filter state for logging purposes. The metadata will be
   // added to StreamInfo's filter state under the namespace corresponding to the ext_authz filter
   // name.
+  //
+  // If this is unset and ``emit_filter_state_stats`` is false, the filter will not emit any filter
+  // state.
   google.protobuf.Struct filter_metadata = 28;
+
+  // When set to true, the filter will emit per-stream stats for access logging. The filter state
+  // key will be the same as the filter name.
+  //
+  // If using Envoy GRPC, emits latency, bytes sent / received, upstream info, and upstream cluster
+  // info. If not using Envoy GRPC, emits only latency.
+  //
+  // If this is false and ``filter_metadata`` is unset, the filter will not emit any filter state.
+  bool emit_filter_state_stats = 29;
```

Comment:
Should we just `emit_filter_state_stats` as single source of control on whether to log stats or not?

filter_metadata logging is part of filter state logging ( 
filter_state->setData(...
                          std::make_shared<ExtAuthzLoggingInfo>(config_->filterMetadata()),)

This will make the control logic simpler and cleaner.

Reply:
see https://github.com/envoyproxy/envoy/pull/35698/files#r1729341640

In other words, do we see realistic use case that people will set `filter_metadata` but not set `emit_filter_state_stats`?